components/raftstore-v2/src/operation/life.rs

```diff
@@ -0,0 +1,281 @@
+    /// 2. Received a message whose target peer's ID is larger than this;
+    /// 3. Applied a conf remove self command.
+    /// In all cases, the peer will be destroyed asynchronousely in next
+    /// handle_raft_ready.
+    /// `triggered_msg` will be sent to stor fsm after destroy is finished.
+    /// Should set the message only when the target peer is supposed to be
+    /// created afterward.
+    pub fn mark_for_destroy(&mut self, triggered_msg: Option<Box<RaftMessage>>) {
+        if self.serving() {
+            self.destroy_progress_mut().wait_with(triggered_msg);
+            self.set_has_ready();
+        }
+    }
+
+    /// In v2, it's possible to destroy the peer without waiting for apply. But
+    /// we better wait till all previous entries are applied in case there
+    /// are split. It's a waste to use snapshot to restore newly split
+    /// tablet.
+    #[inline]
+    pub fn postpond_destroy(&self) -> bool {
+        let entry_storage = self.storage().entry_storage();
+        // TODO: check actual split index instead of commit index.
+        entry_storage.applied_index() != entry_storage.commit_index()
+    }
+
+    /// Start the destroy progress. It will write `Tombstone` state
+    /// asynchronously.
+    ///
+    /// After destroy is finished, `finish_destroy` should be called to clean up
+    /// memory states.
+    pub fn start_destroy(&mut self, write_task: &mut WriteTask<EK, ER>) {
+        let entry_storage = self.storage().entry_storage();
+        if self.postpond_destroy() {
+            return;
+        }
+        let first_index = entry_storage.first_index();
+        let last_index = entry_storage.last_index();
+        if first_index <= last_index {
+            write_task.cut_logs = match write_task.cut_logs {
+                None => Some((first_index, last_index)),
+                Some((f, l)) => Some((cmp::min(first_index, f), cmp::max(last_index, l))),
+            };
+        }
+        let mut extra_states = ExtraStates::new(entry_storage.apply_state().clone());
+        let mut region_state = self.storage().region_state().clone();
+        // Write worker will do the clean up when meeting tombstone state.
+        region_state.set_state(PeerState::Tombstone);
+        extra_states.set_region_state(region_state);
+        extra_states.set_raft_state(entry_storage.raft_state().clone());
+        write_task.extra_write.set_v2(extra_states);
+        self.destroy_progress_mut().start();
+    }
+
+    /// Do clean up for destroy. The peer is permanently destroyed when
+    /// Tombstone state is persisted. This method is only for cleaning up
+    /// memory states.
+    pub fn finish_destroy<T>(&mut self, ctx: &mut StoreContext<EK, ER, T>) {
+        info!(self.logger, "peer destroyed");
+        ctx.router.close(self.region_id());
+        if let Some(msg) = self.destroy_progress_mut().finish() {
+            // It's a best effort.
+            let _ = ctx.router.send_raft_message(msg);
```

Comment:
It will be sent to store fsm to create a new peer.

Reply:
But for send_raft_message, will it first try to send a peer message? And that peer is already closed by "ctx.router.close(self.region_id());" above. 
And according to send_raft_message below,  because sending a peer message failed (as closed), it will return without sending a store msg. 

pub fn send_raft_message(
        &self,
        msg: Box<RaftMessage>,
    ) -> std::result::Result<(), TrySendError<Box<RaftMessage>>> {
        let id = msg.get_region_id();
        let peer_msg = PeerMsg::RaftMessage(msg);
        let store_msg = match self.try_send(id, peer_msg) {
            Either::Left(Ok(())) => return Ok(()),
            Either::Left(Err(TrySendError::Full(PeerMsg::RaftMessage(m)))) => {
                return Err(TrySendError::Full(m));
            }
            Either::Left(Err(TrySendError::Disconnected(PeerMsg::RaftMessage(m)))) => {
                return Err(TrySendError::Disconnected(m));
            }
            Either::Right(PeerMsg::RaftMessage(m)) => StoreMsg::RaftMessage(m),
            _ => unreachable!(),
        };
        match self.send_control(store_msg) {
            Ok(()) => Ok(()),
            Err(TrySendError::Full(StoreMsg::RaftMessage(m))) => Err(TrySendError::Full(m)),
            Err(TrySendError::Disconnected(StoreMsg::RaftMessage(m))) => {
                Err(TrySendError::Disconnected(m))
            }
            _ => unreachable!(),
        }
    }
}

Reply:
If no mailbox is found, `Either::Right` is returned.